### ISSUE TITLE
Popup does not support ignoresSafeArea()/edge-to-edge content

### ISSUE THREAD

**Issue by ryanmitchener** [JetBrains/compose-multiplatform#3741]
**Describe the bug**
Popup does not allow the caller to create a popup that is full screen if edge-to-edge content is enabled on Android and iOS.

**Affected platforms**
- iOS
- Android

**Versions**
- Kotlin version*: 1.9.0
- Compose Multiplatform version*: 1.5.0
- OS version(s)* (required for Desktop and iOS issues): iOS 16
- OS architecture (x86 or arm64): arm64

**To Reproduce**
Steps and/or the code snippet to reproduce the behavior:
```swift
// ContentView.swift
import UIKit
import SwiftUI
import composeUI

struct ComposeView: UIViewControllerRepresentable {
    func makeUIViewController(context: Context) -> UIViewController {
        MainKt.MainViewController()
    }

    func updateUIViewController(_ uiViewController: UIViewController, context: Context) {}
}

struct ContentView: View {
    var body: some View {
        ComposeView()
                .ignoresSafeArea(.all) // Compose has own keyboard handler
    }
}
```

```kotlin
// Main.kt
fun MainViewController(): UIViewController {
    return ComposeUIViewController {
        Box(modifier = Modifier.fillMaxSize().background(Color.Blue))
       
        Popup {
            Box(modifier = Modifier
                .fillMaxSize()
                .background(Color.Green.copy(alpha = .75f))
                .windowInsetsPadding(WindowInsets.systemBars)
            ) {
                Text("Hello")
            }
        }

    }
}
```

**Expected behavior**
If the root view controller has `.ignoresSafeArea()` the popup should fill the entire screen.

**Screenshots**
![Simulator Screenshot - iPhone 14 Pro - 2023-09-28 at 22 37 44](https://github.com/JetBrains/compose-multiplatform/assets/6433145/072ae462-4212-49ed-92a3-ef97da04c617)

**Additional context**
This problem can be worked around using the internal compose API:

```kotlin
@OptIn(InternalComposeApi::class)
fun MainViewController(): UIViewController {
    return ComposeUIViewController {
        Box(modifier = Modifier.fillMaxSize().background(Color.Blue))

        val existingInsets = LocalSafeAreaState.current

        CompositionLocalProvider(LocalSafeAreaState provides remember { mutableStateOf(IOSInsets()) }) {
            Popup {
                CompositionLocalProvider(LocalSafeAreaState provides existingInsets) {
                    Box(
                        modifier = Modifier
                            .fillMaxSize()
                            .background(Color.Green.copy(alpha = .75f))
                            .windowInsetsPadding(WindowInsets.systemBars)
                    ) {
                        Text("Hello")
                    }
                }
            }
        }
    }
}
```

![Simulator Screenshot - iPhone 14 Pro - 2023-09-28 at 22 54 11](https://github.com/JetBrains/compose-multiplatform/assets/6433145/130f0dcf-29d3-420a-8dcd-8eba6317938f)

This also affects Android with `WindowCompat.setDecorFitsSystemWindows(false)` but there isn't as simple of a work around.

**Comment by MatkovIvan** [comment]
It's already solved in 1.5.10-beta02, please look at https://github.com/JetBrains/compose-multiplatform-core/pull/833

> If the root view controller has .ignoresSafeArea() the popup should fill the entire screen.

It's possible by unsetting `usePlatformInsets` flag on iOS, but not the default behaviour.

> This problem can be worked around using the internal compose API:

jfyi this code will not work after proper fix mentioned above in new versions, please use a new flag.

> This also affects Android with WindowCompat.setDecorFitsSystemWindows(false) but there isn't as simple of a work around.

As mentioned in the template - for Android, report it in the [Jetpack Compose tracker](https://issuetracker.google.com/issues/new?component=612128)

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-3741">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
